File: python/Arrays/largest_element.py
def largest_element(arr):
    """
    Find the largest element in an array
    Time Complexity: O(n)
    Space Complexity: O(1)
    """
    if not arr:
        return None
    
    largest = arr[0]
    for i in range(1, len(arr)):
        if arr[i] > largest:
            largest = arr[i]
    return largest

def main():
    # Read input
    n = int(input())
    
    # Read array elements
    arr = list(map(int, input().split()))
    
    # Find largest element
    result = largest_element(arr)
    
    # Print result
    print(result)

if __name__ == "__main__":
    main() 

File: cpp/Arrays/LargestElement.cpp
#include <iostream>
#include <vector>
using namespace std;

int LargestElement(vector<int>& vec)
{
    int largest = vec[0];
    for(int i=1; i<vec.size(); i++) {
        if(vec[i] > largest) {
            largest = vec[i];
        }
    }
    return largest;
}

int main()
{
    int n;
    cin >> n;

    vector<int> vec(n);
    for (int i = 0; i < n; i++)
    {
        cin >> vec[i];
    }

    int result = LargestElement(vec);

    cout << result << endl;
    return 0;
}
